Objects/Converters/ConverterRevit/ConverterRevitShared/Partial Classes/ConvertRebar.cs

```diff
@@ -1,175 +1,348 @@
-        return appObj;
+        startHook = RebarHookToNative(speckleStartHook);
+        Enum.TryParse(speckleStartHook.orientation, out startHookOrientation);
       }
 
-      if (docObj != null)
+      var speckleEndHook = speckleRebar.endHook as RevitRebarHook;
+      RebarHookType endHook = null;
+      RebarHookOrientation endHookOrientation = RebarHookOrientation.Right;
+      if (speckleEndHook != null)
       {
-        rebar = (DB.Structure.Rebar)docObj;
-        // if the number of curves don't match, we need to create a new bar
-        if (rebar.Quantity != closedCurves.Count + openCurves.Count)
-          rebar = null;
+        endHook = RebarHookToNative(speckleEndHook);
+        Enum.TryParse(speckleEndHook.orientation, out endHookOrientation);
       }
 
-      // update curves if we can, only available in revit 2022
-#if REVIT2022
-      if (rebar != null)
-        rebar.MoveBarInSet(0, DB.Transform.Identity);
-#endif
+      // get the shape curves
+      List<DB.Curve> curves = barShape.curves.SelectMany(o => CurveToNative(o).Cast<DB.Curve>()).ToList();
+      if (curves is null || !curves.Any())
+      {
+        appObj.Update(status: ApplicationObject.State.Failed, logItem: "Could not convert any shape curves");
+        return appObj;
+      }
 
-      // create freeform from curveloops if they exist
-      if (curveLoops.Count > 0)
+      // get the rebar plane norm from the curves
+      XYZ normal = XYZ.BasisZ;
+      if (speckleRebar.normal is not null)
       {
-        var result = RebarFreeFormValidationResult.Success;
-        rebar ??= DB.Structure.Rebar.CreateFreeForm(Doc, barType, host, curveLoops, out result);
-        if (result != RebarFreeFormValidationResult.Success)
-        {
-          appObj.Update(status: ApplicationObject.State.Failed, logItem: "Freeform Rebar could not be created from closed curves.");
-          return appObj;
-        }
+        normal = VectorToNative(speckleRebar.normal);
       }
-      else if (openCurves.Count > 0)
+
+      // create the rebar
+      DB.Structure.Rebar rebar = null;
+      try
       {
-        //rebar ??= DB.Structure.Rebar.CreateFromCurves(Doc, barType, host, curveLoops, out result);
+        rebar = DB.Structure.Rebar.CreateFromCurves(
+          Doc,
+          barStyle,
+          barType,
+          startHook,
+          endHook,
+          CurrentHostElement,
+          normal,
+          curves,
+          startHookOrientation,
+          endHookOrientation,
+          true,
+          true
+        );
       }
-      else
+      catch (Exception e)
       {
-        appObj.Update(status: ApplicationObject.State.Failed, logItem: "No convertible rebar curves where found.");
-        return appObj;
+        appObj.Update(status: ApplicationObject.State.Failed, logItem: e.Message);
       }
+      finally
+      {
+        if (rebar != null)
+        {
+          SetInstanceParameters(rebar, speckleRebar);
```

Comment:
This seems pretty strange. If creating the rebar throws an exception, then we're still checking if the rebar was created and then continuing to set parameters. Did you actually experience cases where this would happen?

Reply:
Ah this was left over from when I was handling specific exceptions - will update to return in the catch block